app/code/Magento/CatalogGraphQl/Model/Resolver/Categories/DataProvider/Category/CollectionProcessorInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Categories\DataProvider\Category;
+
+use Magento\Catalog\Model\ResourceModel\Category\Collection;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\GraphQl\Model\Query\ContextInterface;
+
+/**
+ * Add additional joins, attributes, and clauses to a category collection.
+ */
+interface CollectionProcessorInterface
+{
+    /**
+     * Process collection to add additional joins, attributes, and clauses to a category collection.
+     *
+     * @param Collection $collection
+     * @param SearchCriteriaInterface $searchCriteria
+     * @param ContextInterface|null $context
+     * @return Collection
+     */
+    public function process(
+        Collection $collection,
+        SearchCriteriaInterface $searchCriteria,
+        ContextInterface $context = null
```

Comment:
just a minor observation in comparison to the product processor
Noticed that there we passed attributes as well. I guess here we did not need it here. Not sure because this is meant as an extension point if we need to add it and just not use it ourselves..maybe other extensions will..

Reply:
Good point. Now, I have edited to pass attributesNames as well. Currently are not used, but someday maybe someone extends this.